### ISSUE TITLE
`to md` works weirdly on sparse tables

### ISSUE THREAD

**Issue by kaathewisegit** [nushell/nushell#11196]
### Describe the bug

When there is a table with a lot of null values, `to md` discards a bunch of rows.

### How to reproduce

```Nushell
~> ls | append {a: 1} | to md --pretty 
| a |
| - |
| [table 3 rows] |
| 1 |

```

### Expected behavior

I'd probably expect something along the lines of:

```Nushell
| name | type | size | modified                                         | a |
| ---- | ---- | ---- | ------------------------------------------------ | - |
| dirs | dir  | 40 B | Thu, 30 Nov 2023 23:30:13 +0300 (33 seconds ago) |   |
| here | dir  | 40 B | Thu, 30 Nov 2023 23:30:16 +0300 (31 seconds ago) |   |
| some | dir  | 40 B | Thu, 30 Nov 2023 23:30:09 +0300 (38 seconds ago) |   |
|      |      |      |                                                  | 1 |
```

### Configuration

| key                | value                                                        |
| ------------------ | ------------------------------------------------------------ |
| version            | 0.86.0                                                       |
| branch             |                                                              |
| commit_hash        |                                                              |
| build_os           | linux-x86_64                                                 |
| build_target       | x86_64-alpine-linux-musl                                     |
| rust_version       | rustc 1.72.1 (d5c2e9c34 2023-09-13) (Alpine Linux 1.72.1-r0) |
| cargo_version      | cargo 1.72.1                                                 |
| build_time         | 2023-10-22 21:58:37 +00:00                                   |
| build_rust_channel | release                                                      |
| allocator          | mimalloc                                                     |
| features           | default, sqlite, trash, which, zip                           |
| installed_plugins  |                                                              |

**Comment by KAAtheWiseGit** [comment]
And to put some more stuff in here, I don't think that the header and blockquote key behavior is specified anywhere.  And it only works for half of the headers.

**Comment by amtoine** [comment]
was able to reproduce and yeah it looks ugly :laughing:

**Comment by KAAtheWiseGit** [comment]
@amtoine I have a suggestion: I think the header-making capabilities of `to md` should be removed.  I feel that a plugin is better suited for that kind of work.  This would also allow to remove the `htmlescape` library from `nu-command` crate, meaning that the resulting binary (without the extras flag) will be smaller.

**Comment by KAAtheWiseGit** [comment]
@amtoine nope, I only suggest removing the item part.  `to md` stays, but it only accepts tables/records and only outputs Markdown tables.  So `version | transpose key value | to md` works, but `{h1: "I am creating a Markdown file, not a table"} | to md` doesn't.

**Comment by amtoine** [comment]
@KAAtheWiseGit 
aah yeah i see what you mean!

that sounds pretty sensible to me.
not sure how much breaking such a change would be because i don't know how many people would use `to md` for something that's more than a simple md table :thinking: 

also there is the `--per-element` switch that helps when the table contains a header

**Comment by weirdan** [comment]
This now works as expected:

```nushell
ls | take 3 | append {a: 1} | to md --pretty | clip copy
```
```markdown
| name               | type | size    | modified                                       | a   |
| ------------------ | ---- | ------- | ---------------------------------------------- | --- |
| CITATION.cff       | file | 812 B   | Sat, 23 Aug 2025 00:25:21 +0200 (5 months ago) |     |
| CODE_OF_CONDUCT.md | file | 3.4 kB  | Sat, 23 Aug 2025 00:25:21 +0200 (5 months ago) |     |
| CONTRIBUTING.md    | file | 18.3 kB | Thu, 5 Feb 2026 02:23:21 +0100 (3 days ago)    |     |
|                    |      |         |                                                | 1   |
```
